### PR TITLE
Update Account mod

### PR DESCRIPTION
**Pull request by futuretech6**

### CODE REVIEW

File: pkg/account/handler.go [hunk 1]
```diff
@@ -27,33 +41,67 @@ type AccountHandler struct {
...
-	if ok, _ := regexp.MatchString(`^\w+@\w+[.\w+]+$`, Email); !ok {
+		Type   AcountType `json:"type" validate:"required"`
+		Name   string     `json:"name" validate:"required"`
+		Passwd string     `json:"passwd" validate:"required"`
+	}
+
+	var body RequestBody
+
+	if err := c.Bind(&body); err != nil {
+		return c.JSON(http.StatusNotFound, api.Return("Bind Error", nil))
```
Comment (AsterNighT) [inline pkg/account/handler.go]: use a `http.StatusBadRequest` for that

File: pkg/account/handler.go [hunk 2]
```diff
@@ -27,33 +41,67 @@ type AccountHandler struct {
...
+	}
+	if err := c.Validate(&body); err != nil {
+		return c.JSON(http.StatusNotFound, api.Return("Validate Error", nil))
```
Comment (AsterNighT) [inline pkg/account/handler.go]: ditto

File: pkg/account/handler.go [hunk 3]
```diff
@@ -71,18 +119,57 @@ func (h *AccountHandler) CreateAccount(c echo.Context) error {
...
-	Passwd := c.QueryParam("Passwd")
+	type RequestBody struct {
+		Email  string `json:"email" validate:"required"`
+		Passwd string `json:"passwd" validate:"required"`
+	}
+	var body RequestBody
 
-	if ok, _ := regexp.MatchString(`^\w+@\w+[.\w+]+$`, Email); !ok {
-		return c.JSON(http.StatusBadRequest, api.Return("Invali E-mail Address", nil))
+	if err := c.Bind(&body); err != nil {
```
Comment (AsterNighT) [inline pkg/account/handler.go]: This is very much duplicated, can we reuse this part?

File: pkg/account/handler.go [hunk 4]
```diff
@@ -137,48 +224,97 @@ func (h *AccountHandler) ResetPasswd(c echo.Context) error {
...
+	token := jwt.NewWithClaims(jwt.SigningMethodHS256, jwt.MapClaims{
+		"id": u.ID,
+	})
+
+	tokenString, err := token.SignedString(jwtKey)
+	return tokenString, err
+}
+
+// Authoriszed : Check Auth
+func Authoriszed(next echo.HandlerFunc) echo.HandlerFunc {
```
Comment (AsterNighT) [inline pkg/account/handler.go]: typo

Comment (AsterNighT) [inline pkg/account/handler.go]: You need to fix this middleware and register it to echo.

File: pkg/account/model.go [hunk 5]
```diff
@@ -40,3 +42,7 @@ type Patient struct {
 	Cases   []cases.Case `gorm:"foreignkey:ID"`
 	Chats   []chat.Chat  `gorm:"foreignkey:ID"`
 }
+
+var (
+	jwtKey = os.Getenv("JWT_KEY")
```
Comment (AsterNighT) [inline pkg/account/model.go]: you can provide a mock key for debugging

**Review by AsterNighT — CHANGES_REQUESTED** [review]
BTW, there are check errors and muse errors. Pls do check them.

File: pkg/account/handler.go [hunk 6]
```diff
@@ -137,48 +224,97 @@ func (h *AccountHandler) ResetPasswd(c echo.Context) error {
...
+			return jwtKey, nil
+		})
+
+		if !token.Valid || err != nil {
+			return c.NoContent(http.StatusUnauthorized)
+		}
+
+		c.Set("username", token.Claims.(jwt.MapClaims)["username"])
+
+		return next(c)
```
Comment (ziyue-pan) [inline pkg/account/handler.go]: So we can get `username` from `c echo.context` right?

Comment (futuretech6) [reply pkg/account/handler.go]: something like this, am working on it

File: pkg/account/README.md [hunk 7]
```diff
@@ -0,0 +1,78 @@
+# Create
+
+Request
+
+```json
+{
+    "id": 1234567,  // uint
```
Comment (AsterNighT) [inline pkg/account/README.md:24]: json does not accept comments. Consider moving it some where else, like mapping it to a go struct.

**Comment by AsterNighT** [comment]
Well, ok? Just fix the checks.

**Review by AsterNighT — APPROVED** [review]

**Pull request merged**